scripts/write_deployinst_input.py

```diff
@@ -188,66 +187,124 @@ def read_template(template):
-        name = data['reactor_name'].decode('utf-8').title()
-        start = name.find('(')
-        end = name.find(')')
-        if start != -1 and end != -1:
-            name = name[:start]
-        if data['type'].decode('utf-8') == 'BWR':
-            reactor_body = template.render(
+    template_path = '../templates/[reactor]_template_cyborg.xml.in'
+
+    if not is_cyborg:
+        template_path = template_path.replace('_cyborg', '')
+
+    pwr_template = read_template(template_path.replace('[reactor]', 'pwr'))
+    mox_reactor_template = read_template(template_path.replace('[reactor]', 'mox'))
+    candu_template = read_template(template_path.replace('[reactor]', 'candu'))
+
+    ap1000_spec = {'template' : pwr_template,
+                   'assem_size': 446.0,
+                   'n_assem_core': 157,
+                   'n_assem_batch': 52}
+    bwr_spec = {'template' : pwr_template,
+                   'assem_size': 180,
+                   'n_assem_core': 764 / 1000,
+                   'n_assem_batch': 764 / 3000}
+    phwr_spec = {'template' : candu_template,
+                   'assem_size': int(8000 / 47g3),
+                   'n_assem_core': 473 / 500,
+                   'n_assem_batch': 60}
+    candu_spec = {'template' : candu_template,
+                   'assem_size': int(8000 / 473),
+                   'n_assem_core': 473 / 500,
+                   'n_assem_batch': 60}
+    pwr_spec = {'template' : pwr_template,
+                   'assem_size': 446.0,
+                   'n_assem_core': 193 / 1000,
+                   'n_assem_batch': 193 / 3000}
+    epr_spec = {'template' : pwr_template,
+                   'assem_size': 467.0,
+                   'n_assem_core': 216,
+                   'n_assem_batch': 72}
+
+    reactor_specs = {'AP1000' : ap1000_spec,
+                    'PHWR': phwr_spec,
+                    'BWR': bwr_spec,
+                    'CANDU': candu_spec,
+                    'PWR': pwr_spec,
+                    'EPR': epr_spec}
+
+    for data in data_list:
+        # refine name string
+        name = refine_name(data['reactor_name'])
+        reactor_type = data['type'].decode('utf-8')
+        if reactor_type in reactor_specs.keys():
+            # if the reactor type matches with the pre-defined dictionary,
+            # use the specifications in the dictionary.
+            reactor_str = reactor_type
+            spec_dict = reactor_specs[reactor_str]
+            reactor_body = spec_dict['template'].render(
                 country=data['country'].decode('utf-8'),
+                type=reactor_str,
                 reactor_name=name,
-                assem_size=180,
-                n_assem_core=int(round(data['capacity']/1000 * 764)),
-                n_assem_batch=int(round(data['capacity']/3000 * 764)),
+                assem_size=spec_dict['assem_size'],
+                n_assem_core=int(round(spec_dict['n_assem_core'] * data['capacity'])),
```

Comment:
rated capacity? electric? thermal? capacity factor? What is meant by capacity here. please use a more informative variable name.